feature_engine/selection/eliminate_feature_by_feature_importance.py

```diff
@@ -0,0 +1,234 @@
+        Desired metric to optimise the performance for the estimator. Comes from
+        sklearn.metrics. See the model evaluation documentation for more options:
+        https://scikit-learn.org/stable/modules/model_evaluation.html
+
+    threshold: float, int, default = 0.01
+        The value that defines if a feature will be kept or removed. Note that for
+        metrics like roc-auc, r2_score and accuracy, the thresholds will be floats
+        between 0 and 1. For metrics like the mean_square_error and the
+        root_mean_square_error the threshold will be a big number.
+        The threshold must be defined by the user.
+
+    cv : int, default=3
+        Desired number of cross-validation fold to be used to fit the estimator.
+
+    Attributes
+    ----------
+
+
+    Methods
+    -------
+
+    fit: finds important features
+
+    transform: removes non-important / non-selected features
+
+    fit_transform: finds and removes non-important features
+
+    """
+
+    def __init__(
+        self,
+        estimator=RandomForestClassifier(),
+        scoring="roc_auc",
+        cv=3,
+        threshold=0.01,
+        variables=None,
+        regression=False
+    ):
+
+        if not isinstance(cv, int) or cv < 1:
+            raise ValueError("cv can only take positive integers bigger than 1")
+
+        if not isinstance(threshold, (int, float)):
+            raise ValueError("threshold can only be integer or float")
+
+        if not isinstance(regression, bool):
+            raise ValueError("regression should be a boolean")
+
+        self.variables = _define_variables(variables)
+        self.estimator = estimator
+        self.scoring = scoring
+        self.threshold = threshold
+        self.cv = cv
+        self.regression = regression
+
+    def fit(self, X, y):
+        """
+
+        Args
+        ----
+
+        X: pandas dataframe of shape = [n_samples, n_features]
+           The input dataframe
+
+        y: array-like of shape (n_samples)
+           Target variable. Required to train the estimator.
+
+
+        Returns
+        -------
+
+        self
+        """
+
+        # check input dataframe
+        X = _is_dataframe(X)
+
+        # reset the index
+        X = X.reset_index(drop=True)
+        y = y.reset_index(drop=True)
+
+        # find numerical variables or check variables entered by user
+        self.variables = _find_numerical_variables(X, self.variables)
+
+        # train model with all features and cross-validation
+        model = cross_validate(
+            self.estimator,
+            X,
+            y,
+            cv=self.cv,
+            scoring=self.scoring,
+        )
+
+        # store initial model performance
+        self.initial_model_performance_ = model["test_score"].mean()
+
+        # get performance metric
+        scorer = get_scorer(self.scoring)
+
+        # choose the approriate attibute to use for the appropriate model
+        if (self.regression):
```

Comment:
we can get the attributes from the estimators directly like this:
https://github.com/scikit-learn/scikit-learn/blob/0fb307bf3/sklearn/feature_selection/_from_model.py#L18